pkg/search/postgres/common.go

```diff
@@ -268,31 +308,102 @@ func (p *parsedPaginationQuery) AsSQL() string {
 	return paginationSB.String()
 }
 
-func populateGroupBy(querySoFar *query, schema *walker.Schema, queryFields map[string]searchFieldMetadata) {
-	// Explicit GROUP BY clauses are not supported yet. If a query field (in select or order by) is a derived field requiring a group by clause,
+func populateSelect(querySoFar *query, schema *walker.Schema, querySelect *v1.QuerySelect, queryFields map[string]searchFieldMetadata) error {
+	if len(querySelect.GetFields()) == 0 {
+		return nil
+	}
+
+	for _, field := range querySelect.GetFields() {
+		fieldMetadata := queryFields[field]
+		dbField := fieldMetadata.baseField
+		if dbField == nil {
+			return errors.Errorf("field %s in SELECT clause does not exist in table %s or connected tables", field, schema.Table)
+		}
+
+		querySoFar.SelectedFields = append(querySoFar.SelectedFields, pgsearch.SelectQueryField{
+			SelectPath: qualifyColumn(dbField.Schema.Table, dbField.ColumnName, ternary.String(dbField.SQLType == "uuid", "::text", "")),
```

Comment:
This is a really curious line. My understanding is that idiomatically go as a language doesn't want to support ternary because it's deemed difficult to parse. https://go.dev/doc/faq#Does_Go_have_a_ternary_form

Reply:
I believe this function simply encapsulates if-then-else construct under the hood and nothing more.